didcomm_messaging/resolver/didweb.py

```diff
@@ -0,0 +1,162 @@
+
+        # Get the hostname & port
+        hostname = did_segments[2].lower()
+        hostname = hostname.replace("%3a", ":")
+
+        # Resolve the path portion of the DID, if there is no path, default to
+        # a .well-known address
+        path = ".well-known"
+        if len(did_segments) > 3:
+            path = "/".join(did_segments[3:])
+
+        # Assemble the URI
+        did_uri = f"https://{hostname}/{path}/did.json"
+
+        return did_uri
+
+    async def is_resolvable(self, did: str) -> bool:
+        """Determine if the did is a valid did:web did that can be resolved."""
+        if DID.is_valid(did) and did_web_pattern.match(did):
+            return True
+        return False
+
+    @staticmethod
+    def did_from_url(url: str) -> DID:
+        """Convert a URL into a did:web did."""
+
+        # Make sure that the URL starts with a scheme
+        if not url.startswith("http"):
+            url = f"https://{url}"
+
+        # Parse it out to we can grab pieces
+        parsed_url = urlparse(url)
+
+        # Assemble the domain portion of the DID
+        did = "did:web:%s" % parsed_url.netloc.replace(":", "%3A")
+
+        # Cleanup the path
+        path = parsed_url.path.replace(".well-known/did.json", "")
+        path = path.replace("/did.json", "")
+
+        # Add the path portion of the did
+        if len(path) > 1:
+            did += path.replace("/", ":")
+        return did
+
+    @staticmethod
+    async def from_did_peer(did: DID, did_web: DID) -> dict[str, Any]:
+        """Convert a did:peer did into a did:web document."""
+        peer2 = Peer2()
+        peer4 = Peer4()
+        resolved = None
+
+        # Resolve the did:peer to a document
+        if await peer2.is_resolvable(did):
+            resolved = await peer2.resolve(did)
+        if await peer4.is_resolvable(did):
+            resolved = await peer4.resolve(did)
+
+        if not resolved:
+            raise DIDResolutionError("Failed to resolve did:peer:2 or did:peer:4")
+
+        return await DIDWeb._fix_doc(resolved, did_web)
+
+    @staticmethod
+    async def _fix_doc(doc: dict[str, Any], did_web: DID) -> dict[str, Any]:
+        # Replace the did for did:peer with did:web
+        did = doc["id"]
+        doc["id"] = did_web
+        for method in doc["verificationMethod"]:
+            method["controller"] = did_web
+
+        # Perform replacements on embedded keys
+        for relationship in (
+            "authentication",
+            "assertionMethod",
+            "keyAgreement",
+            "capabilityInvocation",
+            "capabilityDelegation",
+        ):
+            method = doc.get(relationship, [])
+            for key in method:
+                if not (isinstance(key, dict) and key.get("controller", None)):
+                    continue
+                key["controller"] = did_web
+
+        # Add the did:peer as an AKA address
+        doc["alsoKnownAs"].append(did)
+        return doc
```

Comment:
By extension (on my last comment), I believe this method should also be omitted.